packages/frontend/src/components/common/Table/BodyCell.tsx

```diff
@@ -65,22 +66,40 @@ const BodyCellWrapper: FC<CommonBodyCellProps> = ({ onSelect, ...props }) => {
 
     const [isCellSelected, setIsCellSelected] = useState<boolean>(false);
 
+    const canHaveContextMenu = !!CellContextMenu && props.hasData;
+
     const handleCellSelect = useCallback(
         (cellId: string | undefined) => {
+            if (!canHaveContextMenu) return;
+
             onSelect?.(cellId);
             setIsCellSelected(cellId ? true : false);
         },
-        [onSelect, setIsCellSelected],
+        [onSelect, setIsCellSelected, canHaveContextMenu],
     );
 
-    const canHaveContextMenu = !!CellContextMenu && props.hasData;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Hook deps seem fine. Do we need this?

Reply:
since its not wrapped inside inline method it still warns. i'll try to get rid of it